packages/nextra-theme-docs/src/components/flexsearch.tsx

```diff
@@ -99,46 +87,166 @@ const loadIndexesImpl = async (
   let pageId = 0
 
   for (const [route, structurizedData] of Object.entries(searchData)) {
-    let pageContent = ''
-    ++pageId
+    const sections: Section[] = []
 
     for (const [key, content] of Object.entries(structurizedData.data)) {
-      const [headingId, headingValue] = key.split('#')
-      const url = route + (headingId ? '#' + headingId : '')
-      const title = headingValue || structurizedData.title
-      const paragraphs = content.split('\n')
-
-      sectionIndex.add({
-        id: url,
-        url,
+      const [anchor, title] = key.split('#', 2)
+
+      sections.push({
+        anchor,
         title,
-        pageId: `page_${pageId}`,
-        content: title,
-        ...(paragraphs[0] && { display: paragraphs[0] })
+        content: content.replace(/\s+/g, ' ')
```

Comment:
```suggestion
        content
```
this replace is better to add to remark plugin https://github.com/shuding/nextra/blob/62eb7436bfd3fec9aa7a75d0cca03127c9ae4444/packages/nextra/src/server/mdx-plugins/remark-structurize.ts#L26-L34